### PR TITLE
Resurrect framian pt. 1

### PR DESCRIPTION
**Pull request by longcao**
ultralight beaming

This is a lil bit of work to try and get framian in a better place:

* Drop `2.10.x` support (cross publishing, macros)
* Kill `framian-json`
* Revbump `shapeless` from `2.0.0` => `2.3.0` (everything seems to be fine locally (JDK 8)... tests pass)
* Update travis.yml to not cross build and use jdk7/8 (unsure if this will actually pass... Travis is still seems finicky)

There's more I wish to do but didn't want to let the PR get too large:

* Change `organization` and the source headers to be not-Pellucid (your choice here)
* Fix publishing (perhaps use [sbt-release](https://github.com/sbt/sbt-release) here) and publish under whatever `organization`, perhaps not-Bintray as well
* Fix ScalaDoc publishing, maybe using a combination of [sbt-ghpages](https://github.com/sbt/sbt-ghpages) and [sbt-site](https://github.com/sbt/sbt-site) and maybe pushing it to the `*.github.io` domain?
* Bump sbt plugins (and perhaps figure out what's going on here... https://github.com/tixxit/framian/blob/0494ee1a1eb5adb69c1f51bf552ef636daeca7f4/project/plugins.sbt#L6-12)
* Fast-forward compile/test dependencies where possible, fix compilation issues that will arise from skipping lots of version numbers. I wasn't able to drop in spire 0.11.0 from 0.8.2 due to some compile errors that I didn't want to tackle just yet.
* `specs2` => `ScalaTest` if that's desireable (I know it is by me :D, but I remember you saying you were thinking about doing this a long time ago, seems doable but a largely mechanical task)

:pray: Please let me know if I missed anything or if you'd like something else not mentioned above. :pray:

### CODE REVIEW

**Comment by longcao** [comment]
Broken jdk7 buid on Travis appears to be a common thing: https://github.com/travis-ci/travis-ci/issues/5227

**Comment by longcao** [comment]
(also https://github.com/travis-ci/travis-ci/issues/5669)

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
+  - 2.11.8
 jdk:
-- oraclejdk7
-- openjdk7
+  - oraclejdk8
+  - oraclejdk7
+  - openjdk7
 install:
-- ./.travis-cache.sh
+  - ./.travis-cache.sh
```
Comment (tixxit) [inline .travis.yml]: Can we delete this and see what happens? We can do the sudo: false thing to trigger the docker builds which should use some sort of caching for SBT or something.

**Comment by tixxit** [comment]
This all looks great. If you don't mind the `net.tixxit` org for now, I have that set up for publishing on sonatype and can add the release pluging tonight. I'll probably just copy the config over from [delimited](https://github.com/tixxit/delimited) which is set up for publishign scaladocs and releasing to sonatype already.

Definitely :+1: to specs2 => scalatest

**Comment by tixxit** [comment]
Feel free to merge.

File: .travis.yml (hunk 1 above)
Comment (longcao) [inline .travis.yml]: Done! Let's see how Travis response to this.

**Comment by longcao** [comment]
@tixxit cool - I'll leave the publishing and doc build settings unfiddled-with so you can drop in your stuff to do so later.

**Pull request merged**